### PR TITLE
Add bind data to auto-populate bind form

### PR DESCRIPTION
**Pull request by sedroche**
Ticket: https://issues.jboss.org/browse/AEROGEAR-2413

Adding new property on metadata that will be used to auto-populate binding params form.
 
Setting a type of `path` will retrieve the value from the resource.
Setting a type of `default` will set that value
```.json
 '{"name": "CLIENT_ID", "value": "metadata.name", "type": "path"}'
 '{"name": "CLIENT_TYPE", "value": "public", "type": "default"}'
```

### Verification
Use this UI branch -> https://github.com/aerogear/origin-web-console/pull/38

#### Client binding
custom-client-auto

Client Id should be filled.

### CODE REVIEW

File: apb.yml [hunk 1]
```diff
@@ -17,7 +17,9 @@ plans:
   - name: default
     description: This default plan deploys custom-runtime-connector-apb
     free: True
-    metadata: {}
+    metadata:
+      MobileClient_bind_parameters_data:
```
Comment (dimitraz) [inline apb.yml]: It might look more consistent to make the resource kind all lowercase, just my opinion though. wdyt?

**Review by dimitraz — APPROVED** [review]
Verified locally :+1:

File: apb.yml (hunk 1 above)
Comment (sedroche) [reply apb.yml]: Yeah I think that makes sense. I'll add that in. Cheers.

**Review by dimitraz — APPROVED** [review]
Verified that the Client ID is correctly pre-filled :+1:

**Review by JameelB — APPROVED** [review]

**Pull request merged**